core/services/workflows/syncer/handler.go

```diff
@@ -499,6 +486,61 @@ func (h *eventHandler) workflowRegisteredEvent(
 	return nil
 }
 
+// getWorkflowArtifacts retrieves the workflow artifacts from the database if they exist,
+// or downloads them from the specified URLs if they are not found in the database.
+func (h *eventHandler) getWorkflowArtifacts(
+	ctx context.Context,
+	payload WorkflowRegistryWorkflowRegisteredV1,
+) ([]byte, []byte, error) {
+	spec, err := h.orm.GetWorkflowSpec(ctx, hex.EncodeToString(payload.WorkflowOwner), payload.WorkflowName)
+	if err != nil {
+		return h.fetchAndDecodeWorkflowArtifacts(ctx, payload)
+	}
+
+	// there is an update in the BinaryURL or ConfigURL, lets fetch it
+	if spec.ConfigURL != payload.ConfigURL || spec.BinaryURL != payload.BinaryURL {
```

Comment:
Since you're operating on the workflowID and it is the same, you actually wouldn't need this check.

The flow should basically be either get the artifacts from the DB and if that fails for any reason recreate it. I don't think you need anything else.